### PR TITLE
Query params

### PR DESCRIPTION
**Pull request by martyall**
Allows for query params (from servant) to be part of query URL

### CODE REVIEW

File: hs-abci-test-utils/src/Tendermint/Utils/Client.hs [hunk 1]
```diff
@@ -36,40 +39,75 @@ instance RunClient (ReaderT RPC.Config IO) where
...
   , clientResponseRaw  :: Resp.Query
   }
 
+addQueryParamsToPath
+  :: QueryStringList
+  -> Text
+  -> Text
+addQueryParamsToPath qs path =
+  let qParams = intercalate "&" $ map (\(n,v) -> n <> "=" <> v) qs
+  in path <> "?" <> qParams
```
Comment (IvantheTricourne) [inline hs-abci-test-utils/src/Tendermint/Utils/Client.hs]: This seems to be adding `?` even though there aren't any `qs`.

Comment (IvantheTricourne) [reply hs-abci-test-utils/src/Tendermint/Utils/Client.hs]: For example: https://travis-ci.com/f-o-a-m/hs-abci/jobs/276480230#L1304

File: hs-abci-sdk/test/Tendermint/SDK/Test/SimpleStorage.hs [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+          updateCount (Count updateCountTxCount)
+
+--------------------------------------------------------------------------------
+-- Server
+--------------------------------------------------------------------------------
+
+type CountStoreContents = '[(CountKey, Count)]
+
+type GetMultipliedCount =
+     "multiply"
```
Comment (IvantheTricourne) [inline hs-abci-sdk/test/Tendermint/SDK/Test/SimpleStorage.hs]: Probably mean `multiplier`, as indicated by the tests: https://github.com/f-o-a-m/hs-abci/pull/194/files#diff-f3e7ffe5a8bd3efb979298b272d06c93R46

**Comment by IvantheTricourne** [comment]
So it seems that the full path to the query isn't being properly populated when it gets here: https://github.com/f-o-a-m/hs-abci/pull/194/files?file-filters%5B%5D=.hs#diff-18418e639e179d33970efb24cc9030f1R63

I snuck in a log to see what the path was that was triggering the `Path not found.` exception, and it looks like this:

```
queryPath = "multiplied?factor=4"
```

File: hs-abci-sdk/test/Tendermint/SDK/Test/QuerySpec.hs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+              , txGas = 0
+              , txSignature = undefined
+              , txSignBytes = undefined
+              , txSigner = undefined
+              , txNonce = undefined
+              }
+        txContext <- T.newTransactionContext tx
+        _ <- SS.evalToIO ctx . T.eval txContext $ handler tx
+        let q = Req.Query
+              -- TODO -- this shouldn't require / count
```
Comment (IvantheTricourne) [inline hs-abci-sdk/test/Tendermint/SDK/Test/QuerySpec.hs:45]: I'm guessing that this isn't the case anymore?

File: hs-abci-sdk/test/Tendermint/SDK/Test/SimpleStorage.hs [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+-- Server
+--------------------------------------------------------------------------------
+
+type CountStoreContents = '[(CountKey, Count)]
+
+type GetMultipliedCount =
+     "manipulated"
+  :> Capture "subtract" Integer
+  :> QueryParam' '[Required, Strict] "factor" Integer
+  :> BaseApp.Leaf Count
```
Comment (IvantheTricourne) [inline hs-abci-sdk/test/Tendermint/SDK/Test/SimpleStorage.hs:128]: Question: Aside from `Leaf`, how is this different from how servant normally does things?

**Review by IvantheTricourne — APPROVED** [review]

File: hs-abci-sdk/test/Tendermint/SDK/Test/QuerySpec.hs (hunk 3 above)
Comment (martyall) [reply hs-abci-sdk/test/Tendermint/SDK/Test/QuerySpec.hs:45]: yeah but I'm gonna merge it anyway

File: hs-abci-sdk/test/Tendermint/SDK/Test/SimpleStorage.hs (hunk 4 above)
Comment (martyall) [reply hs-abci-sdk/test/Tendermint/SDK/Test/SimpleStorage.hs:128]: Servant has the option for `GET, PUT, POST` etc, we don't have any of these but we still need to specify the return type

**Pull request merged**